File: src/api/presence.ts
import { RequestFN } from "~/core/request-function";
import { TabLocation, type SessionHandle } from "~/models";

export const presence = async (session: SessionHandle): Promise<void> => {
  const request = new RequestFN(session, "Presence", {
    _Signature_: { onglet: TabLocation.Presence }
  });

  await request.send();
};

File: library/src/commonMain/kotlin/ink/literate/pawnote/api/presence.kt
package ink.literate.pawnote.api

import ink.literate.pawnote.core.RequestFN
import ink.literate.pawnote.models.SessionHandle
import ink.literate.pawnote.models.TabLocation
import kotlinx.serialization.encodeToString
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.buildJsonObject
import kotlinx.serialization.json.put
import kotlinx.serialization.json.putJsonObject

suspend fun presence(session: SessionHandle) {
  val request =
      RequestFN(
          session.information,
          "Presence",
          Json.encodeToString(
              buildJsonObject {
                putJsonObject("_Signature_") { put("onglet", TabLocation.Presence.code) }
              }))

  request.send()
}
